internal/page/page.go

```diff
@@ -74,3 +79,33 @@ func (o limitOffset) Int64() int64 {
 func (o limitOffset) String() string {
 	return strconv.Itoa(int(o))
 }
+
+type Metadata struct {
+	Previous string `json:"previous,omitempty"`
+	Next     string `json:"next,omitempty"`
+}
+
+func MetadataResponse(ctx context.Context, r *http.Request, total int) *Metadata {
+	limit := Limit(ctx).Int()
+	offset := Offset(ctx).Int()
+	nextOffset := math.Min(total, offset+limit)
+	prevOffset := math.Max(0, offset-limit)
+
+	q := r.URL.Query()
+
+	// set the "prev" link with offset - limit or 0
+	q.Set("limit", strconv.Itoa(limit))
+	q.Set("offset", strconv.Itoa(prevOffset))
+	params, _ := url.PathUnescape(q.Encode())
+	prev := fmt.Sprintf("%v?%v", r.URL.Path, params)
```

Comment:
I'm not sure if there is no URL native way to format the path (copy the URL object and manipulate it for prev and next) insted of doing the string transpolation.
I guess it's quite safe, but I'd rather see native way, it's easy to make a mistake here :shrug:

Reply:
I tried the naive way, but I think this is a good approavh and safer one. 
I saw that sources used that way.

Reply:
I'm not sure we  want to strictly follow what Sources are doing :innocent:

Reply:
I agree :) but I find it better I guess but I can change it if you prefer.

Reply:
nah, it's fine :) If you like this better, lets keep it, not a blocker for sure! :+1: